### PR TITLE
Add abstraction for plotter data

### PR DESCRIPTION
**Pull request by eurydice76**
See #68 for info.

### CODE REVIEW

File: Src/GUI/Plugins/PlotterPlugin.py [hunk 1]
```diff
@@ -469,28 +472,40 @@ def on_load_data(self, event=None):
...
+                else:
+                    data[vname]['axis'] = []
+                data[vname]['path'] = vpath
+                data[vname]['data'] = arr
+                data[vname]['units'] = getattr(variable, "units", "au")
+
+            unique_name = self.unique(basename, self.plugin._dataDict)
+        
+            self.plugin._dataDict[unique_name]={'data': data, 'path': filename, 'basename': basename}
+            self.plugin._dataPanel.show_dataset(-1)
```
Comment (RastislavTuranyi) [inline Src/GUI/Plugins/PlotterPlugin.py:508]: I can't see the netcdf file being closed; am I wrong? Ideally, I would love if we could implement a context manager to handle the file, to ensure that it is always closed.

File: Src/GUI/Plugins/PlotterPlugin.py [hunk 2]
```diff
@@ -128,9 +130,10 @@ def build_panel(self):
             self.datalist = wx.ListCtrl(self.setup, wx.ID_ANY,style = wx.LC_REPORT|wx.LC_SINGLE_SEL)
 
         self.datalist.InsertColumn(0, 'Variable', width=100)
-        self.datalist.InsertColumn(1, 'Axis', width=50)
-        self.datalist.InsertColumn(2, 'Dimension')
-        self.datalist.InsertColumn(3, 'Size')
+        self.datalist.InsertColumn(1, 'Path', width=100)
```
Comment (RastislavTuranyi) [inline Src/GUI/Plugins/PlotterPlugin.py:133]: I would like to propose a different order for these variables; I think that keeping the order as it has been and just adding the path at the end (last column) is better, given that I don't know  what knowing the path could be useful for. What do you think?

**Review by RastislavTuranyi — COMMENTED** [review]
Hi, thank you for your work! I think the new naming works better, though I have a couple extra comments. The main issue is the file not being closed. Theoretically,  a context manager would be best, but I'm not sure if it is practical in this particular case;, so I'll take a look at this, if you don't mind.

File: Src/GUI/Plugins/PlotterPlugin.py [hunk 3]
```diff
@@ -469,28 +472,40 @@ def on_load_data(self, event=None):
...
+                    break
+            else:
+                continue
+
+            data = collections.OrderedDict()
+            for vname, vinfo in f.variables.items():
+                vpath, variable = vinfo
+                arr = variable.get_array()
+
+                if not numpy.issubdtype(arr.dtype, numpy.number):
```
[the remark this answers is not in the document]
Comment (eurydice76) [reply Src/GUI/Plugins/PlotterPlugin.py:489]: yeap you right the extra `numpy.issubdtype(arr.dtype, numpy.number)` is superfluous as the non numeric variables are already filtered out.

File: Src/GUI/Plugins/PlotterPlugin.py (hunk 1 above)
Comment (eurydice76) [reply Src/GUI/Plugins/PlotterPlugin.py:508]: yeap a context manager would the best thing to do

**Comment by eurydice76** [comment]
thanks for the quick review Rastislav. As you will investigate for the context manager, I let you also proceed to the small changes you recommended in your review.

File: Src/GUI/Plugins/PlotterData.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                var_key = '{:s}_{:d}'.format(var_key, comp)
+                comp += 1
+
+            var_dict[var_key] = (path, NetCDFPlotterVariable(var))
+
+        for _, sub_group in group.groups.items():
+            NetCDFPlotterData.find_numeric_variables(var_dict, sub_group)
+
+
+PLOTTER_DATA_TYPES = {'.nc': NetCDFPlotterData, '.cdf': NetCDFPlotterData, '.netcdf': NetCDFPlotterData}
```
Comment (RastislavTuranyi) [inline Src/GUI/Plugins/PlotterData.py:129]: I changed this dict to make accessing the class easier/more readable at the cost of repetition in this definition. I think this should be ok as long as no class has too many extensions, but feel free to change it if you prefer a different solution.

**Comment by RastislavTuranyi** [comment]
Also, I keep getting this error:
```
Exception RuntimeError: RuntimeError(u'NetCDF: Not a valid ID',) in <bound method NetCDFPlotterData.__del__ of <MDANSE.GUI.Plugins.PlotterData.NetCDFPlotterData object at 0x000000003B9C5DC8>> ignored
```
which causes no issues (since it's ignored), but I have not idea why this is happening.

File: Src/GUI/Plugins/PlotterData.py (hunk 4 above)
Comment (eurydice76) [reply Src/GUI/Plugins/PlotterData.py:129]: no that's OK for me

**Comment by eurydice76** [comment]
Hi @RastislavTuranyi , I removed the __del__ method of _IPlotterData class and it seems to solve the error/warning that you get. I think that we are ready to merge.

**Review by RastislavTuranyi — APPROVED** [review]
I can certainly confirm that the warning is gone.

**Comment by eurydice76** [comment]
Hi @sanghamitra-mukhopadhyay , could you please review and proceed to merge if you are OK ?

**Review by sanghamitra-mukhopadhyay — APPROVED** [review]

**Pull request merged**